Sources/MessageIndex.php

```diff
@@ -328,7 +329,8 @@ function MessageIndex()
 		' . (!empty($message_index_tables) ? implode("\n\t\t\t\t", $message_index_tables) : '') . '
 		WHERE t.id_board = {int:current_board} '
 			. (!$modSettings['postmod_active'] || $context['can_approve_posts'] ? '' : '
-			AND (t.approved = {int:is_approved}' . ($user_info['is_guest'] ? '' : ' OR t.id_member_started = {int:current_member}') . ')') . '
+			AND (t.approved = {int:is_approved}' . ($user_info['is_guest'] ? '' : ' OR t.id_member_started = {int:current_member}') . ')') . (!empty($message_index_topic_wheres) ? ' 
+			AND ' . implode("\n\t\t\t\tAND ", $message_index_topic_wheres) : ''). '
```

Comment:
Like I mentioned before I decline here any changes from author,
Since the part is fragile and important.

Reply:
Yeah, it is fragile, however, we shouldn't limited what mod authors can do based on that logic.

We aim to offer freedom and options, we can't and shouldn't decide how those option will be used.

Reply:
Will @DiegoAndresCortes fix all performance and error issue which get raised up in the support board of sm.org?

Reply:
The only issue that directly impacts a vanilla installation is directly applying the where var without making sure it has some value first. I already asked for that change.

It is not our duty to provide support for poorly performant mods. It is not our duty to limit what mods can do. Our duty is to allow mod authors the freedom to modify SMF.